cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KRaftMigrationUtils.java

```diff
@@ -0,0 +1,126 @@
+     * @param zkConnectionString    Connection string to the ZooKeeper ensemble to connect to
+     */
+    public static void deleteZooKeeperControllerZnode(Reconciliation reconciliation, Secret clusterCaCertSecret, Secret coKeySecret, long operationTimeoutMs, String zkConnectionString) {
+        PasswordGenerator pg = new PasswordGenerator(12);
+        // Setup truststore from PEM file in cluster CA secret
+        // We cannot use P12 because of custom CAs which for simplicity provide only PEM
+        String trustStorePassword = pg.generate();
+        File trustStoreFile = Util.createFileTrustStore(KRaftMigrationUtils.class.getName(), "p12", Ca.certs(clusterCaCertSecret), trustStorePassword.toCharArray());
+
+        // Setup keystore from PKCS12 in cluster-operator secret
+        String keyStorePassword = new String(Util.decodeFromSecret(coKeySecret, "cluster-operator.password"), StandardCharsets.US_ASCII);
+        File keyStoreFile = Util.createFileStore(KRaftMigrationUtils.class.getName(), "p12", Util.decodeFromSecret(coKeySecret, "cluster-operator.p12"));
+        try {
+            ZooKeeperAdmin admin = createZooKeeperAdminClient(reconciliation, zkConnectionString, operationTimeoutMs,
+                    trustStoreFile, trustStorePassword, keyStoreFile, keyStorePassword);
+            admin.delete("/controller", -1);
+            admin.close();
+            LOGGER.infoCr(reconciliation, "KRaft migration rollback ... /controller znode deleted");
+        } catch (IOException | InterruptedException | KeeperException ex) {
+            LOGGER.warnCr(reconciliation, "Failed to delete /controller znode", ex);
+        } finally {
+            if (trustStoreFile != null) {
+                if (!trustStoreFile.delete())   {
+                    LOGGER.warnCr(reconciliation, "Failed to delete file {}", trustStoreFile);
+                }
+            }
+            if (keyStoreFile != null)   {
+                if (!keyStoreFile.delete())   {
+                    LOGGER.warnCr(reconciliation, "Failed to delete file {}", keyStoreFile);
+                }
+            }
+        }
+    }
+
+    /**
+     * Check for the status of the Kafka metadata migration
+     *
+     * @param reconciliation    Reconciliation information
+     * @param kafkaAgentClient  KafkaAgentClient instance to query the agent endpoint for migration state metric
+     * @param controllerPodName Name of the quorum controller leader pod
+     *
+     * @return true if the migration is done, false otherwise
+     */
+    public static boolean checkMigrationInProgress(Reconciliation reconciliation, KafkaAgentClient kafkaAgentClient, String controllerPodName) {
+        KRaftMigrationState kraftMigrationState = kafkaAgentClient.getKRaftMigrationState(controllerPodName);
+        LOGGER.infoCr(reconciliation, "ZooKeeper to KRaft migration state {} checked on controller {}", kraftMigrationState.state(), controllerPodName);
```

Comment:
I think it's useful to provide information about migration status even at INFO level. Any suggestion to make it less _debugy_?

Reply:
Moved to DEBUG level because it's a more lower level logging information (about what the Kafka metrics are returning via the Kafka agent). We already have logging at INFO level for the user to know what's going on about migration.